predicators/pybullet_helpers/robots/single_arm.py

```diff
@@ -319,6 +343,32 @@ def forward_kinematics(self, joint_positions: JointPositions) -> Pose3D:
         position = ee_link_state.worldLinkFramePosition
         return position
 
+    def _validate_joints_state(self, joint_positions: JointPositions,
+                               target_pose: Pose3D) -> None:
+        """Validate that the given joint positions matches the target pose.
+
+        This method should NOT be used during simulation mode as it
+        resets the joint states.
+        """
+        # Store current joint positions so we can reset
+        initial_joint_states = self.get_joints()
+
+        # Set joint states, forward kinematics to determine EE position
+        self.set_joints(joint_positions)
+        ee_pos = self.get_state()[:3]
+        target_pos = target_pose
+        pos_is_close = np.allclose(ee_pos,
+                                   target_pos,
+                                   atol=CFG.pybullet_ik_tol)
```

Comment:
Keep in mind IK tol for IKFast is different (and non-existent), so maybe we want to add another setting for this (or not I think it might be sufficient)

Reply:
it should always pass this check though right? IKFast should be very close all the time?

Reply:
In theory I think it should. But might be worth still validating if we want to be certain.